### ISSUE TITLE
[java] MissingOverride doesn't detect missing @Override

### ISSUE THREAD

**Issue by krichter722** [pmd/pmd#1852]
**Affects PMD Version:** 6.15.0

**Rule:** `category/java/bestpractices.xml/MissingOverride`

**Description:**
A missing `@Override` annotation like in

```
public class ClassX implements Runnable {

    public void run() {
        System.out.println("run");
    }
}
```

isn't detected if PMD is configured as shown in the code sample. I expect the code sample to fail because `run` in `ClassX` overrides `run` defined in `Runnable`.

**Code Sample demonstrating the issue:**
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>de.richtercloud</groupId>
    <artifactId>pmd-missing-override</artifactId>
    <version>1.0-SNAPSHOT</version>
    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.0</version>
                <configuration>
                    <source>11</source>
                    <target>11</target>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
                <version>3.12.0</version>
                <executions>
                    <execution>
                        <id>pmd-check</id>
                        <phase>validate</phase>
                        <goals>
                            <goal>check</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <printFailingErrors>true</printFailingErrors>
                    <rulesets>
                        <ruleset>code-analysis/pmd-ruleset.xml</ruleset>
                    </rulesets>
                    <analysisCache>false</analysisCache>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>net.sourceforge.pmd</groupId>
                        <artifactId>pmd-core</artifactId>
                        <version>6.15.0</version>
                    </dependency>
                    <dependency>
                        <groupId>net.sourceforge.pmd</groupId>
                        <artifactId>pmd-java</artifactId>
                        <version>6.15.0</version>
                    </dependency>
                </dependencies>
            </plugin>
        </plugins>
    </build>
</project>
```
together with the `pmd-ruleset.xml`

```
<?xml version="1.0"?>
<ruleset name="Custom ruleset"
        xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
    <description>
        The default ruleset
    </description>
    <rule ref="category/java/bestpractices.xml/MissingOverride" />
</ruleset>
```

A reproducer can be found at https://gitlab.com/krichter/pmd-missing-override/ and CI output at https://gitlab.com/krichter/pmd-missing-override/-/jobs/223478332.

**Running PMD through:** *Maven*

**Comment by adangel** [comment]
@krichter722 Thanks for the excellent bug report!

In the gitlab build output, I don't see that the maven-pmd-plugin is executed at all.
I think, the culprit is `<phase>validate</phase>` - the correct phase should be `verify`.

Validate is running at the very first step (so, I'm still puzzled why maven-pmd-plugin is not run at all here, would have expected it to be one of the first steps), verify is running after the project has been built. See [Maven Lifecycle](http://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html).

And "verify" is the correct phase for pmd (it's also the [default phase](https://maven.apache.org/plugins/maven-pmd-plugin/check-mojo.html), so you don't need to specify it at all): PMD needs the compiled project to work on, to determine the type of the classes - and especially in this rule, we use the type to see, whether a method is inherited (and overridden) or not...

**Comment by krichter722** [comment]
@adangel Thanks for your comment. I consider myself an advanced user of PMD, however never figured out that `verfiy` is the correct phase to run PMD. Thanks for that particular very helpful piece of information as well.

The code in the reproducer on GitLab is different from the one in the code example section due to a copy-paste-error for which I'm sorry. In the reproducer there's no `executions` section.
